### PR TITLE
Xeno reworkish, Slimes now have feelings!

### PR DESCRIPTION
**Pull request by jcatjmeow**
### Intent of your Pull Request
First off, this does the following:
- Slimes aren't on lavaland anymore.
- Slime mutation rate is up to 30% (from 5%)
- Slimes have feelings now

What does that last one mean? Well.
Slimes now have a personality value, and a happiness value. Slimes are telepathic social creatures, and so they like being near other slimes. They don't like being lonely, or near slimes that hate them, however. If you put two slimes together with opposite personalities they will both become very unhappy.

An unhappy slime does not produce slime cores. Additionally, any time it reproduces or splits, the children have a chance to be stillborns, who also won't produce any useful cores.

You can tell a slime's personality by how it reacts to other slimes. Slimes that hate each other will try to run away from each other, and slimes that like each other will try to go closer to each other.

When a slime does split (assuming of course, it was not a stillbirth) it will completely re-invent it's personality and happiness level. So if you don't have slimes that like each other you can try to reproduce them to get ones that do!

The mapped-in slimes start at the max happiness, to have a little "buffer" in case you get bad rng, so you don't brickwall instantly. Baby slimes start at the default level however, which is unhappy but not super unhappy.

You can tell a slime's happiness level by looking at it. A happy slime will have a :3 face! A sad slime will have a :( face instead.

A slime needs to have a happy face to produce cores, but a slime that is decently close to happy but has a sad face will not produce stillborns. Only a very unhappy slime will give birth to dead slimes.

Oh, and the process of dying makes slimes a little bit unhappy. The quicker you kill them the better. Be humane!

Note: I tested and tweaked these values a lot. There is probably room for more testing, however! It does work bug free as far as I know.
#### Changelog

:cl:  
rscdel: Slimes are gone from lavaland
tweak: Slime mutation chance has gone way up
tweak: Slimes need to be around a slime friend to be happy and make cores. Details on how to do this in the pr!
wip: This is work in progress. 
imageadd: amazing sprite from Partheo for the ruined slime extract! It's so gross!

/:cl:

### CODE REVIEW

**Comment by Ratvar** [comment]
Does this address issues of xenobiology loot being op, or issue of xenobiologists never leaving xenobiology beyond printing parts and console they researched, or a part of such prs - if not, I would like if slimes were rarer by miners dying, xenobiologists being too bored to wait and mutation chance being low, as they currenty are

**Review by Partheo — APPROVED** [review]

**Comment by jcatjmeow** [comment]
> Does this address issues of xenobiology loot being op, or issue of xenobiologists never leaving xenobiology beyond printing parts and console they researched, or a part of such prs - if not, I would like if slimes were rarer by miners dying, xenobiologists being too bored to wait and mutation chance being low, as they currenty are

No. But I feel like this makes xeno more fun to do. 
I can always make the happiness go down by more if we feel like the slimes are happy too easily. It would need testing I feel. Or I could lower the mutation rate to like 10% or 15% or so. Maybe make it scale off of happiness? I'm open for ideas

**Comment by Ratvar** [comment]
I believe making xenobio fun is bad because xenobio is bad, and I think that making xenobio happen more often will make more people see how dumb it is, and how it needs to be gutted and reworked - support

**Comment by jcatjmeow** [comment]
> I believe making xenobio fun is bad because xenobio is bad, and I think that making xenobio happen more often will make more people see how dumb it is, and how it needs to be gutted and reworked - support

ok yeah but if I make a "remove xenobio" pr it'll get instaclosed I need to make it a buff/nerf combo that still leaves it functional or people will riot

**Comment by Jumps0** [comment]
Good luck trying to run away from eachother in my 1x1 pens

**Review by SomeguyManperson — APPROVED** [review]
yes

**Comment by Jarod1200** [comment]
unhappy slimes should vocally express their displeasure with some sort of annoying sound

**Comment by jcatjmeow** [comment]
> unhappy slimes should vocally express their displeasure with some sort of annoying sound

get me a nice sound and I'll do it

**Comment by Titanslayer71** [comment]
I tried to breed them as I usually did in Xenobio.

They all gave me ruined Grey Slime Extract, and I used up all the monkey cubes I had.

Now all my slimes are dead, and I don't have monkey cubes to continue feeding them!

**Comment by Titanslayer71** [comment]
> unhappy slimes should vocally express their displeasure with some sort of annoying sound

I don't play ss13 with sound either though. o-o

**Comment by SomeguyManperson** [comment]
might need to make grey slimes immune to this so xenobio doesn't shoot themself in the foot because they haven't learned the system

File: code/modules/mob/living/simple_animal/slime/death.dm [hunk 1]
```diff
@@ -4,11 +4,18 @@
 	if(!gibbed)
 		if(is_adult)
 			var/mob/living/simple_animal/slime/M = new(loc, colour)
+			M.happiness = 0
+			if(!ckey && happiness < 0 && prob(abs(happiness)*2))//so at -20 its a 40% chance. at -50 happiness it's an 100% chance
+				M.death()
+				M.desc += " It looks like it was a stillborn..."
```
Comment (SomeguyManperson) [inline code/modules/mob/living/simple_animal/slime/death.dm]: this won't show up due to how examining mobs works, you'll need to make it work with the examine proc

**Comment by Titanslayer71** [comment]
I left some of the slimes for a few minutes... now all of them are dead! o-o

**Comment by jcatjmeow** [comment]
> might need to make grey slimes immune to this so xenobio doesn't shoot themself in the foot because they haven't learned the system

Map-spawned slime start at max happiness to give xenobios a buffer.

If they become that unhappy that quickly it's more than just a grey slime issue. >.>

File: code/modules/mob/living/simple_animal/slime/death.dm (hunk 1 above)
Comment (jcatjmeow) [reply code/modules/mob/living/simple_animal/slime/death.dm]: > I left some of the slimes for a few minutes... now all of them are dead! o-o

Slimes don't die from being unhappy. Did you put them in a cold room?

Comment (jcatjmeow) [reply code/modules/mob/living/simple_animal/slime/death.dm]: Ah fuck i put that in the wrong place...

@SomeguyManperson it does show up tho >.>

Comment (Titanslayer71) [reply code/modules/mob/living/simple_animal/slime/death.dm]: No! They're all dead in their pen! [https://gyazo.com/a11a94a7319301c046e0607455798dce](url)

Png link fixed.

**Comment by jcatjmeow** [comment]
IDEA:

The first two slimes that spawn automatically have the same personality value. So as long as you start by putting them together, nothing can go wrong!

**Comment by Titanslayer71** [comment]
> IDEA:
> 
> The first two slimes that spawn automatically have the same personality value. So as long as you start by putting them together, nothing can go wrong!

At least give us more monkey cubes, and emergency grey slime extract to create a new grey slime...

I don't even have slimes, or monkey cubes to restart Xenobio anymore!

**Comment by Jumps0** [comment]
> > IDEA:
> > The first two slimes that spawn automatically have the same personality value. So as long as you start by putting them together, nothing can go wrong!
> 
> At least give us more monkey cubes, and emergency grey slime extract to create a new grey slime...
> 
> I don't even have slimes, or monkey cubes to restart Xenobio anymore!

or like a cube that makes a single grey baby slime similar to a monkey cube

**Comment by Partheo** [comment]
> 
> 
> unhappy slimes should vocally express their displeasure with some sort of annoying sound

slime noises would be hilarious. happy and unhappy noises, I 100% support this

**Comment by jcatjmeow** [comment]
> > IDEA:
> > The first two slimes that spawn automatically have the same personality value. So as long as you start by putting them together, nothing can go wrong!
> 
> At least give us more monkey cubes, and emergency grey slime extract to create a new grey slime...
> 
> I don't even have slimes, or monkey cubes to restart Xenobio anymore!

This change is meant to make xenobio harder, and make it more involved with more things to manage.

While I think I made it a bit tooooo hard, if you run out of slimes or monkey cubes tough luck. Go do another science job

**Comment by jcatjmeow** [comment]
btw mapspawned slimes (like the ones you start with) share a personality. So they will make themselves happy and never fight. 

Once they split, however, it's anyone's game

**Comment by Xantam-13** [comment]
If this was on the game last round, and the round before, the xenobiologists reaaaaaaally didn't like it.

**Comment by jcatjmeow** [comment]
6d57aad should help a lot once it's testmerged.

But regardless, xenobio now has to work a little harder. And actually think about the slimes and use their head a little bit.

**Comment by jcatjmeow** [comment]
Note to self tomarrow: slim down the personality types

**Comment by Xoxeyos** [comment]
Make slime cookies improve moods, not so sure on this PR.

**Comment by boodaliboo** [comment]
> > unhappy slimes should vocally express their displeasure with some sort of annoying sound
> 
> I don't play ss13 with sound either though. o-o

same i usually dont aswell could we get a small visual que?

**Comment by jcatjmeow** [comment]
> same i usually dont aswell could we get a small visual que?

> You can tell a slime's happiness level by looking at it. A happy slime will have a :3 face! A sad slime will have a :( face instead.

**Comment by jcatjmeow** [comment]
Should stillborn slimes drop regular cores or ruined cores?

If they drop regular cores it will be harder to softlock cause dead grays can be recycled

**Comment by boodaliboo** [comment]
maybe make ruined cores have a way to be recovered thats annoying to do like needing tricordazine from chem or something along the lines

**Comment by jcatjmeow** [comment]
> maybe make ruined cores have a way to be recovered thats annoying to do like needing tricordazine from chem or something along the lines

I don't want to add a way to circumvent this pr's mechanics. Otherwise people will

**Comment by MonsWTF** [comment]
PR contains the details and changes, but it doesn't provide the specific _Intent_ of the PR, which is the most important part for any changes to the game, and most particularly Xenobiology. Xenobio, for some reason, has two groups of people that either have massive hate-boners for anyone that doesn't agree with their opinion that Xenobio needs to be deleted, or are the minority ofpeople that do the job, focus on specific extracts/crossbreeds, and then turn to Valid'ing/powergaming.

One of my current concerns I have with any newer PR's for Xenobio is whether or not it addresses the issue of the disparity between new/mildly experienced Xenobio's and "ultimate instinct" Xenobio's. The job in its current form is an autistic mess for the 1% of Xenobiologists that know how to do the job 'correctly'. Whether 30% or 5% starting mutation, Xenobio can get to Rainbow slime by 35 minutes or 50 minutes respectively, without any involvement from Mining. Same goes for getting the majority of slime colors by the 20 minute or 35 minute mark given the same mutation circumstances. 

It's hard for me to be able to accurately comment on the efficacy of the changes here in how that's going to affect the gameplay loop of an experienced Xenobio; depending on how finicky the mood system is, I could see it helping to reduce the meta I personally have gotten used to of "pump out as many slimes as possible and get 'em all to mutate", but I rarely see other Xenobs copying my practice unless I taught them it myself. 

All I can currently say is that it's essentially going to be adding another layer of multitasking into a job that demands constant attention of the user in order to do it well, lest the player give up having any usefulness/involvement in the round at all; if that's the end goal, I'm unsure how happy I'd be with having a role that's job doesn't actually let them engage with others in the station at some point. If the purpose of this PR and the Xenobio-mining PR prior to this one have the same goal of keeping Xenobio in check but still allowing for the colors to appear, what is the reason this will fare better than its predecessor?

Final thoughts: 
1) Thoughts on having slime's mood at time of splitting alter the range for changing mutation %? (default range is -6% to +6% when a slime splits)
2) Thoughts on having slime's mood at time of splitting effect the actual value of Mutation %? (A happy slime w/ 30% mutation splits with 30%, a sad slime w/ 30% mutation splits with 15%, etc.)
3) Did anyone ever port that TG PR that fixes the bug with slimes not doing damage to glass?

**Comment by jcatjmeow** [comment]
@MonsWTF

The intent is to have xeno be nerfed, same as the lavaland slimes change, except that this actually requires paying attention/multitasking etc and is just all around more interesting than "sometimes you find random slimes on lavaland lol"

I can make the mutation chance scale with happiness quite easily. I don't know how well that would be received given the current reception amoung those who playtested this, but yeah

**Comment by jcatjmeow** [comment]
Someone suggested to me that slimes in nicer rooms should feel better. I like the idea on principle since it means you have to design little slime habitats. In practice i don't think it would be much, people would game the system and just scatter the furniture in the pen dwarf fortress style.

Also if anyone has any thoughts on how to make the job more involved with the rest of the station, let me know, I'd love to do that

**Comment by MonsWTF** [comment]
> I can make the mutation chance scale with happiness quite easily. I don't know how well that would be received given the current reception amoung those who playtested this, but yeah

The way I see it is this; with the current PR if the Xenobiologist decides to kill off a slime, or splits them 'incorrectly', they are going to be hit with an irreversible negative reward in both circumstances, and that's not a pleasant experience to have to start all over again or to have to trash whatever you just did.

Altering the rate at which the slimes are able to mutate is a safer thing to experiment with; even if done wrong, you still get cores. It's significantly less-permanent, slows down the rate at which new slime colors are discovered (which is what all of the PR's are trying to balance within Xenobio), and doesn't directly impact the # of cores the scientist loses/receives. The 'core' task of Xenobiology is manipulating the Mutation %, not the cores themselves.

On that note, 30% default is way too high for Xenobio. 10-15% is a better 'early' range for Xenobio to start with and encourages the player to have to work their way up if they want those Rainbows. Even 5% is fine, if the slime's mood can affect the mutation % growth range.

Does the slime console give the player any feedback in how a slime 'feels' about its fellow slime neighbors? Xenobio's regularly have to scan each slime every 2-4 minutes to monitor their Growth Stage and their Mutation %, so it would be good to know if that info scan states the slime's mood, along with values of their "positive/negative neighbors" so that the Xenobiologist has an idea on who needs to move.

I get you want to nerf Xenobio's gameplay loop by requiring more multitasking...but the job already requires a ton of multitasking. There's a reason that Xenobiologists rarely leave their lab and it saddens me that most of the people I see complaining about the job either unknowingly or intentionally ignore that fact; it's already an invested experience. 

What is it **specifically** that you see as a problem of Xenobiology that needs to be nerfed? The first attempted non-specific nerf was removing the console, the second was removing the mutation %, this would be the third. I like to see change, but the proposed changes that have happened previously keep me cautiously optimistic about whether the changes people want to take now are taking into account the job they are impacting.

> Someone suggested to me that slimes in nicer rooms should feel better. 

I agree, sounds interesting in concept but weird or too game-y in practice.

**Comment by jcatjmeow** [comment]
>On that note, 30% default is way too high for Xenobio. 10-15% is a better 'early' range for Xenobio to start with and encourages the player to have to work their way up if they want those Rainbows. Even 5% is fine, if the slime's mood can affect the mutation % growth range.

In my original opinion, the mutation chance could be as high as 50% with this change and it would be fine. The variety of slimes in my mind is a good thing. What I thought the was issue was the _extracts_, not the slimes.

But yeah. 30% was how it used to be before. But it is pretty high. If I lower it down to 10% I'd have to make the other mechanics less punishing. Not sure on the balance, would need some testing, but I am open to it.

**Comment by MonsWTF** [comment]
I can tell ya right now that the original reasons behind previous Xenobio nerfs was that getting the "late-game" slime colors, particularly Adamant, Light Pink, and Rainbow, extremely early on in the round was upsetting some people. Thus, the removal of the consoles, and then the drastic reduction in mutation %. 

With the emotions mechanic, I wouldn't say that the direct impact is specifically to cut back on extracts - you could do that easily with the Slime Grinder and cut back its output rates - but the quantity of slimes that are alive at any given moment, in order to make each slime 'happy' enough to be culled and produce worthwhile extracts. There is still going to be the ability to work around that - the Xenobio could use the additional cages to sufficiently spread out enough slimes so that they can still mass produce the extracts for crossbreeding or personal use - but that's not really addressing the core problem of the extracts themselves.

There is also the concern of making Xenobiology both harder and more micro-intensive than it currently is, which is going to further solidify the 'isolation experience' of the job and fueling complaints that Xenobios don't interact with the rest of the station.

**Comment by jcatjmeow** [comment]
This needs **work**

**Comment by alexkar598** [comment]
This PR has been closed due to being stale. Contact me or another maintainer on Discord if you wish for this PR to be reopened.

**Pull request closed without merging**